### PR TITLE
SVE, Distant Lands, and Alecto Mod Support

### PR DESCRIPTION
**Pull request by Witchybun**
Adds game-side support for the three mods Stardew Valley Expanded, Distant Lands, and Alecto.  

It also includes some tweaks to other mods.
- Reduced the multipliers to the scaling to some modded skills; at default the rates are too high.
- Update mod versions to be recent.

### CODE REVIEW

File: StardewArchipelago/Locations/CodeInjections/Vanilla/RecipePurchaseInjections.cs [hunk 1]
```diff
@@ -321,8 +329,6 @@ private static void AddArchipelagoRecipesToSaloonStock(Dictionary<ISalable, int[
             {
                 AddArchipelagoCookingRecipeItem(saloonStock, "Cookies", 300, myActiveHints);
             }
-
-            AddArchipelagoCookingRecipeItem(saloonStock, "Triple Shot Espresso", 5000, myActiveHints);
```
Comment (agilbert1412) [inline StardewArchipelago/Locations/CodeInjections/Vanilla/RecipePurchaseInjections.cs:325]: Why is the Triple Shot Espresso gone?

**Review by agilbert1412 — CHANGES_REQUESTED** [review]

File: StardewArchipelago/GameModifications/CodeInjections/Modded/SocializingConfigCodeInjections.cs [hunk 2]
```diff
@@ -6,12 +6,12 @@ namespace StardewArchipelago.GameModifications.CodeInjections.Modded
 {
     internal class SocializingConfigCodeInjections
     {
-        private const double TALKING_MULTIPLIER = 2.5;
-        private const double GIFT_MULTIPLIER = 2;
-        private const double EVENT_MULTIPLIER = 2;
-        private const double QUEST_MULTIPLIER = 2;
-        private const double LOVED_GIFT_MULTIPLIER = 1.5;
-        private const double BIRTHDAY_GIFT_MULTIPLIER = 1.5;
+        private const double TALKING_MULTIPLIER = 1.4;
```
[the remark this answers is not in the document]
Comment (Witchybun) [reply StardewArchipelago/GameModifications/CodeInjections/Modded/SocializingConfigCodeInjections.cs:9]: Way too much yes, I would talk to one person and level up on default, way faster than any skill.  Trivialized it.

File: StardewArchipelago/GameModifications/CodeInjections/Modded/ArchaeologyConfigCodeInjections.cs [hunk 3]
```diff
@@ -6,10 +6,10 @@ namespace StardewArchipelago.GameModifications.CodeInjections.Modded
 {
     internal class ArchaeologyConfigCodeInjections
     {
-        private const double ARTIFACT_SPOT_MULTIPLIER = 2; // Default Value: 10
-        private const double DIGGING_MULTIPLIER = 2; // Default Value: 5
-        private const double PANNING_MULTIPLIER = 5; // Default Value: 5
-        private const double WATER_SHIFTER_MULTIPLIER = 2.5; // Default Value: 2
+        private const double ARTIFACT_SPOT_MULTIPLIER = 1.3; // Default Value: 10
```
[the remark this answers is not in the document]
Comment (Witchybun) [reply StardewArchipelago/GameModifications/CodeInjections/Modded/ArchaeologyConfigCodeInjections.cs:9]: Yeah felt it was too much.  I toned it down similarly to Socializing.  I'm more curious how players will feel about the new multiplier for this one.

File: StardewArchipelago/Items/Mail/ModLetterActions.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+using System.Collections.Generic;
+using StardewValley;
+using StardewValley.Tools;
+
+namespace StardewArchipelago.Items.Mail
+{
+    public class ModLetterActions
+    {
+
+        public void ModLetters(Dictionary<string, Action<string>> letterActions)
```
[the remark this answers is not in the document]
Comment (Witchybun) [reply StardewArchipelago/Items/Mail/ModLetterActions.cs:12]: Forgotten I believe.  I re-implemented it.

**Pull request merged**